From 84b1defe99a15b96d32f4c2fecb2c8e9149f696c Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@nxp.com>
Date: Wed, 15 Aug 2018 09:53:07 +0800
Subject: [PATCH] MLK-18947: ASoC: fsl_sai: fix volatile function

The FSL_SAI_VERID and FSL_SAI_PARAM only available
when reg_offset is 8

Signed-off-by: Shengjiu Wang <shengjiu.wang@nxp.com>
(cherry picked from commit 0a0695672dc7ecf07a7642ff6f99f0b9d3a26b32)
---
 sound/soc/fsl/fsl_sai.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

--- a/sound/soc/fsl/fsl_sai.c
+++ b/sound/soc/fsl/fsl_sai.c
@@ -1162,9 +1162,11 @@ static bool fsl_sai_volatile_reg(struct
 	if (reg == FSL_SAI_TCSR(offset) || reg == FSL_SAI_RCSR(offset))
 		return true;
 
+	if (sai->soc->reg_offset == 8 && (reg == FSL_SAI_VERID ||
+				reg == FSL_SAI_PARAM))
+		return true;
+
 	switch (reg) {
-	case FSL_SAI_VERID:
-	case FSL_SAI_PARAM:
 	case FSL_SAI_TFR0:
 	case FSL_SAI_TFR1:
 	case FSL_SAI_TFR2:
